### PR TITLE
Log User Interactions

### LINKED ISSUE #188: Correct Log Interactions with complete timestamp

### PR DESCRIPTION
**Pull request by light-and-salt**
Log user interactions using `angular-websql`
- Log complete timestamp (fix #188)
- Log userid, actionId, actionCategory, label and data
- Export json log file

A lot of code in this PR is recovered from https://github.com/vega/vega-lite-ui/commit/fe903982406ceabcbe2075039ca5bc1747f3c8d8
- [x] npm run lint
- [x] npm run test
- [x] tested with Voyager2

Sample logs:

``` csv
userid,time,actionCategory,actionId,label,data
undefined,Tue Aug 16 2016 01:16:16 GMT-0700 (PDT),DATA,DATASET_CHANGE,Cars,undefined
undefined,Tue Aug 16 2016 01:16:16 GMT-0700 (PDT),POLESTAR,SPEC_CHANGE,"{""data"":{},""mark"":""__ANY__"",""encoding"":{""row"":{},""column"":{},""x"":{},""y"":{},""x2"":{},""y2"":{},""color"":{},""opacity"":{},""size"":{},""shape"":{},""detail"":{},""text"":{},""label"":{},""path"":{},""order"":{},""__ANY__0"":{}},""config"":{}}",undefined
undefined,Tue Aug 16 2016 01:16:17 GMT-0700 (PDT),POLESTAR,SPEC_CHANGE,"{""data"":{""url"":""data/cars.json""},""mark"":""__ANY__"",""encoding"":{""row"":{},""column"":{},""x"":{},""y"":{},""x2"":{},""y2"":{},""color"":{},""opacity"":{},""size"":{},""shape"":{},""detail"":{},""text"":{},""label"":{},""path"":{},""order"":{},""__ANY__0"":{}},""config"":{}}",undefined
```

screen shot 2016-08-16 at 1 18 10 am

### CODE REVIEW

**Comment by kanitw** [comment]
Test failing ..

File: src/services/logger/logger.service.js [hunk 1]
```diff
@@ -84,17 +84,103 @@ angular.module('vlui')
...
         Analytics.trackEvent(action.category, action.id, label, value);
+
+        if (consts.logToWebSql) {
+          var row = {
+            userid: user,
+            time: new Date(),
+            actionCategory: action.category,
+            actionId: action.id,
+            label: _.isObject(label) ? JSON.stringify(label) : label, // zening: do we need label?
+            data: data ? JSON.stringify(data) : undefined // zening: do we log data or value?
```
Comment (kanitw) [inline src/services/logger/logger.service.js]: Hmm, I don't know what's value.  Have you checked if all `logInteraction` call have `value`?  I don't think so.

Comment (kanitw) [inline src/services/logger/logger.service.js]: I think the `value` crap is added randomly in this commit ...  https://github.com/vega/vega-lite-ui/commit/59d59284b3c18e5ae003b9461fe37e9f3f13b386

Comment (light-and-salt) [inline src/services/logger/logger.service.js]: Since `data` contains `value`, I won't log `value`

File: bower.json [hunk 2]
```diff
@@ -40,5 +40,8 @@
     "lodash": "~4.5.1",
     "vega-lite": "1.1.4",
     "selectize": "~0.12.1"
+  },
+  "resolutions": {
+    "angular": "~1.5.7"
```
Comment (kanitw) [inline bower.json:46]: Isn't there a way to fix without resolving?

Comment (kanitw) [inline bower.json:46]: Like actually updating versions of libraries with outdated version.

**Comment by kanitw** [comment]
- [x] Can't I just get the LOG in CSV, which is way easier to analyze?  The old code does export to CSV as well, so I'm not sure why you choose JSON instead.

**Comment by kanitw** [comment]
Plus your JSON is weirdly nested.  At least it should be an array of objects instead object with 1,2,3,4 as keys.

File: bower.json (hunk 2 above)
Comment (light-and-salt) [inline bower.json:46]: `angular-websql` wants `angular` to be `">=1.2.0 <1.5.0"` https://github.com/paulocaldeira17/angular-websql/blob/master/bower.json

Comment (kanitw) [inline bower.json:46]: ok too bad then.

**Pull request merged**